main.js

```diff
@@ -2,43 +2,57 @@
 'use strict';
 
 const config = require('./config.js');
-const extractor = require('./extractor.js');
 const git = require('./git.js');
-const transport = require('./transport.js');
+const scanService = require('./scan-service.js');
+const util = require('util');
+const prettyMilliseconds = require('pretty-ms');
+const { exit } = require('process');
 
-async function generateReport(configuration) {
+async function runScan(configuration) {
     if (!configuration.isValid()) {
         console.error('Invalid configuration used for the reshift-plugin-npm')
+        exit(1);
     }
 
     // lets get the git information for this project folder
     const gitInstance = new git.Git(configuration.projectDir);
+    const repoInfo = await gitInstance.getRepositoryInfo();
     const meta = await gitInstance.branchInfo();
     meta.commitHash = await gitInstance.commitHash();
-
-    var ex = new extractor.Extractor(configuration.projectDir);
-    ex.writeProjectMeta(meta);
-    ex.doTrap();
-    await ex.renameTrap();
-    const compressedFile = await ex.compressOutput();
-
-    const sendMechanism = new transport.Transport();
-    sendMechanism.sendResult(
+    var branch = meta.remotes
+    if (!!!branch) {
```

Comment:
wha?

Reply:
Lol this is short hand for not null, not undefined and not empty :) apparently in javascript they are different lol